### PR TITLE
148 -> v1-release  Text Input uncontrolled

### PR DESCRIPTION
**Pull request by wmangimelli**
- [x] Add nativeAttribute.defaultValue
- [x] Rename displayValue to something more fitting, e.g. internalValue

### CODE REVIEW

**Comment by aVileBroker** [comment]
I think this solution makes it look like an uncontrolled input but it's still controlled. I would think that we should be trying to pass no value to the native input if none was passed into the component - so the browser can do its usual text input stuff and we just report the onChange as usual.

Keeping the defaultValue is important for having uncontrolled inputs though

**Comment by aVileBroker** [comment]
After talking it through - this is actually really close. Just need:

- [ ] rename displayName/setDisplayName as we're no longer displaying it
- [ ] instead of `nativeHTMLAttributes.value || ''`, do `nativeHTMLAttributes.value || nativeHTMLAttributes.defaultValue`

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -186,13 +187,12 @@ const TextInput = ({
   } else if (isMultiline) {
     InputComponent = TextAreaInputContainer;
   }
-  const displayValue = nativeHTMLAttributes.value || defaultValue;
+  const [displayValue, setDisplayValue] = useState(nativeHTMLAttributes.value || '');
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:190]:
```suggestion
  const [internalValue, setInternalValue] = useState(nativeHTMLAttributes.value || nativeHTMLAttributes.defaultValue);
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -204,8 +204,9 @@ const TextInput = ({
               ? e.target.value
               : e.target.value.slice(0, maxLength);
           }
-          if (nativeHTMLAttributes.onChange) {
-            nativeHTMLAttributes.onChange(e);
+          setDisplayValue(e.target.value);
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:207]:
```suggestion
          setInternalValue(e.target.value);
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -144,14 +145,14 @@ const defaultCallback = () => {}; // eslint-disable-line @typescript-eslint/no-e
...
   // Destructure native HTML attributes to provide default values
-  defaultValue = '',
   type = 'text',
   disabled = false,
   cols = 10,
   rows = 10,
 
   debouncedOnChange = defaultCallback,
   onClear,
+  onChange,
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:155]: what's the advantage of pulling this out of the nativeHTMLAttributes?

**Review by aVileBroker — CHANGES_REQUESTED** [review]

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx (hunk 3 above)
Comment (wmangimelli) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx:155]: I had this within before and things went awry. When it was NOT destructed out, customizing the onChange on InputComponent would always get overwritten when spreading the native props at the _end_ of the listed props. We could keep it in the nativeAttributes, ie not destruct it out, and make sure we add our customized props for the Input Component, including onChange, after the spread of the props so the custom props take precedence. I will remove the destructed onChange, and instead move the {...nativeAtrributes} to the first listed prop on InputComponent

Comment (aVileBroker) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx]: I think your way works just fine

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 4]
```diff
@@ -179,35 +173,40 @@ const TextInput = ({
...
       {iconPrefix && createIcon(StyledIconContainer, iconPrefix)}
       <InputComponent
-        value={displayValue}
-        type={type}
-        disabled={disabled}
-        cols={cols}
-        rows={rows}
+        // Set default values above nativeHTMLAttributes
+        type="text"
+        disable={false}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:191]:
```suggestion
        disabled={false}
```

**Review by aVileBroker — CHANGES_REQUESTED** [review]

**Review by aVileBroker — APPROVED** [review]

**Pull request merged**